src/rules/no-duplicate-store-ids.ts

```diff
@@ -0,0 +1,47 @@
+import { createEslintRule } from '../utils/rule-creator'
+import { AST_NODE_TYPES } from '@typescript-eslint/utils'
+
+export const RULE_NAME = 'no-duplicate-store-ids'
+export type MESSAGE_IDS = 'duplicatedStoreIds'
+type Options = []
+
+const usingStoreIds = new Set<string>()
+
+export default createEslintRule<Options, MESSAGE_IDS>({
+  name: RULE_NAME,
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Disallow duplicate store ids.'
+    },
+    schema: [],
+    messages: {
+      duplicatedStoreIds: 'No duplicated store ids allowed.'
+    }
+  },
+  defaultOptions: [],
+  create: (context) => {
```

Comment:
```suggestion
  create: (context) => {
  const usingStoreIds = new Set<string>()
```

Reply:
I think the rule should still work if the context of the store IDS is inside here :)

Reply:
Hi @9romise, sorry for the late reply. I've been travelling for work. I'm not 100% sure eslint rules should work across files, so I need to do some more tests on this in the next days before I merge it.

Reply:
No worries at all! 😊 Thanks for letting me know. It's important to ensure the eslint rules are effective across files. I completely understand your decision to hold off on merging until further testing. Let me know if there's anything I can assist with in the meantime.